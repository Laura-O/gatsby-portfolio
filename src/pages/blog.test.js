import React from 'react';
import { shallow } from 'enzyme';

import Index from './blog';
import {
  sitePropType,
  imagePropType,
  markdownRemarkPropType,
} from '../../test/fixtures';

describe('blog', () => {
  test('rendering', () => {
    const renderProps = {
      data: {
        site: sitePropType,
        allMarkdownRemark: {
          edges: [markdownRemarkPropType],
        },
        indexCover: {
          sizes: imagePropType,
        },
      },
    };
    const wrapper = shallow(<Index {...renderProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('no post', () => {
    const noProps = {
      data: {
        site: sitePropType,
        allMarkdownRemark: {
          edges: [],
        },
        indexCover: {
          sizes: imagePropType,
        },
      },
    };
    const wrapper = shallow(<Index {...noProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
