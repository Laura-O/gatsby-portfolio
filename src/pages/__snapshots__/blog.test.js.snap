// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog no post 1`] = `
<div
  className="blog-posts"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      title
       • Blog
    </title>
  </HelmetWrapper>
</div>
`;

exports[`blog rendering 1`] = `
<div
  className="blog-posts"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      title
       • Blog
    </title>
  </HelmetWrapper>
  <div
    className="blog-post-preview"
    key="id"
  >
    <h3
      className="blog-title"
    >
      <GatsbyLink
        to="path"
      >
        title
      </GatsbyLink>
    </h3>
    <Image
      Tag="div"
      alt=""
      fadeIn={true}
      sizes={
        Object {
          "sizes": "something",
        }
      }
    />
    <small
      className="blog-date"
    >
      date
    </small>
    <div
      className="blog-excerpt"
    >
      excerpt
    </div>
    <div
      className="blog-time"
    >
      10
       mins
    </div>
  </div>
</div>
`;
